src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -618,6 +643,64 @@ public Void get(long timeout, TimeUnit unit) {
 			throw new UnsupportedOperationException();
 		}
 
+		// this as discard hook
+		@Override
+		public void accept(I i) {
+			parent.sourceCleanup.accept(i);
+			// propagates discard to the downstream
+			Operators.onDiscard(i, actual.currentContext());
+		}
+
+		// Context interface impl
+		@Override
+		@SuppressWarnings("unchecked")
+		public <T> T get(Object key) {
+			if (KEY_ON_DISCARD.equals(key)) {
+				return (T) this;
+			}
+
+			return actual.currentContext().get(key);
+		}
+
+		@Override
+		public boolean hasKey(Object key) {
+			if (KEY_ON_DISCARD.equals(key)) {
+				return true;
+			}
+
+			return actual.currentContext().hasKey(key);
+		}
+
+		@Override
+		public Context put(Object key, Object value) {
+			return actual
+					.currentContext()
+					.put(KEY_ON_DISCARD, this)
+					.put(key, value);
+		}
+
+		@Override
+		public Context delete(Object key) {
+			return actual
+					.currentContext()
+					.put(KEY_ON_DISCARD, this)
+					.delete(key);
+		}
+
+		@Override
+		public int size() {
+			return actual.currentContext()
+			             .put(KEY_ON_DISCARD, this)
+			             .size();
+		}
+
+		@Override
+		public Stream<Map.Entry<Object, Object>> stream() {
+			return actual.currentContext()
+			             .put(KEY_ON_DISCARD, this)
+			             .stream();
+		}
+
```

Comment:
The overall put context is a bit convoluted although i understand why you inline Context. Size could be simplified (`size + 1`), stream too (concat),  but for delete and put we might just need to add a putAll or a multi put (varargs ?) in reactor. Other solutions are probably more complicated, but its giving some good idea for Context support like should we support lazy merge to avoid looking up currentContext.

Reply:
My question is - does any downstream supply a context at all? Is it useful to do `Operators.onDiscrd(releasedByteBuf, actual.currectContext())`.

The main question - which subscriber is interested in ByteBuf memory of which is already freed? What one should do with such? We already take care of releasing. Also, we are the last handler in the chain, so should one do anything with such messages after all? (except such messages are Strings)

Reply:
should be fixed